spotify_project/base/views.py

```diff
@@ -23,20 +22,24 @@ def spotify_login(request):
 def spotify_callback(request):
     oauth = OAuth()
     request.session['token_response'] = oauth.get_token_json(request)
+    access_token = request.session['token_response']['access_token']
+    spotify = Spotify(access_token)
+    user = spotify.fetch_current_user()
```

Comment:
It would be worth making a custom authentication backend that does most of this. Then you could use `authenticate` + `login`. It would be more correct. But it's your call.

Reply:
I got so close to accomplishing this. I just can't get login to work.

Reply:
scratch that. I believe I've got it.